frontend/src/components/FilesPage/FilesPage.tsx

```diff
@@ -0,0 +1,126 @@
+import { ChangeEventHandler, useState } from 'react';
+import { Navigate } from 'react-router-dom';
+import { MdFolder, MdOutlineInsertDriveFile, MdSearch } from 'react-icons/md';
+import {
+    Cell,
+    ColumnName,
+    FileRow,
+    FolderRow,
+    HeaderRow,
+    IconWrapper,
+    ListContainer,
+    Name,
+    SortIcon,
+    NameButton,
+    InputWithSpace,
+    BreadcrumbsWithSpace,
+} from './services/styled';
+import {
+    by,
+    formatSize,
+    getBreadcumbs,
+    useCurrentDirectory,
+} from './services/helpers';
+import routes, { generateFolderPath } from '../../routes';
+import iconSizes from '../../styleTokens/iconSizes';
+import PageContainer from '../PageContainer/PageContainer';
+
+const FilesPage = () => {
+    const {
+        isLoading,
+        directory: { data, directoryParts },
+    } = useCurrentDirectory();
+
+    const [inputValue, setInputValue] = useState('');
+
+    const [isDescending, setIsDescending] = useState(false);
+
+    if (isLoading) return null;
+
+    if (!data) return <Navigate replace to={routes.notFound} />;
+
+    // NOTE: Sorting and filtering is an expensive operation on big data sets.
+    // Right now, we don't need a memorization since every component update causes a change in options order or number.
+    // Please verify that every time component functionality changes.
+    const sortedItems = [...data.items]
+        .filter(({ name }) => name.startsWith(inputValue))
+        .sort(by('name'));
+    const orderedSortedItems = isDescending
+        ? sortedItems.reverse()
+        : sortedItems;
+
+    const files = orderedSortedItems.filter(({ type }) => type === 'file');
+    const folders = orderedSortedItems.filter(({ type }) => type === 'dir');
+
+    const handleToogleOrder = () => {
```

Comment:
```suggestion
    const handleToggleOrder = () => {
```

Reply:
fixed in d11ea9c395be0ce966ca4a64440e584339cbe471